src/Controls/src/Core/UriImageSource.cs

```diff
@@ -108,7 +108,7 @@ public override string ToString()
 
 		static string GetCacheKey(Uri uri)
 		{
-			return Device.PlatformServices.GetHash(uri.AbsoluteUri);
+			return Application.Current.Handler.GetRequiredService<IHashAlgorithm>().ComputeHashString(uri.AbsoluteUri);
```

Comment:
The is the "new" way to get services from XAML objects. Not sure if this is the best, but since there should always be an Application and if the app is running, there will always be a handler.

These services will be application services, so you can't get windows and UI things, such as animation managers or UI threads. But, if you are executing code in this area, there may not be a UI object. If you really need it then I suppose you can always go get the Application.Windows and pick a window.